File: JavascriptAlgorithm/DpPlan/I/maxSubArray_53.js
var maxSubArray = function (nums) {
  // -2,1,-3,4,-1,2,1,-5,4 -> 6
  let max = nums[0];
  let maxTill = nums[0];
  for (let i = 1; i < nums.length; i++) {
    if (maxTill + nums[i] > nums[i]) {
      maxTill = maxTill + nums[i];
    } else {
      maxTill = nums[i];
    }
    max = Math.max(max, maxTill);
  }
  return max;
};
export { maxSubArray };

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/WeekContest/D220626/maxSubArray_53.java
package algo.tzashinorpu.WeekContest.D220626;

public class maxSubArray_53 {
    public int maxSubArray(int[] nums) {
        int max = nums[0];
        int n = nums.length;
        int[] dp = new int[n];
        dp[0] = max;
        for (int i = 1; i < n; i++) {
            dp[i] = Math.max(dp[i - 1] + nums[i], nums[i]);
            max = Math.max(dp[i], max);
        }
        return max;
    }
}

